overlord/configstate/configcore/kernel.go

```diff
@@ -0,0 +1,213 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//go:build !nomanagers
+// +build !nomanagers
+
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package configcore
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/snapcore/snapd/asserts"
+	"github.com/snapcore/snapd/i18n"
+	"github.com/snapcore/snapd/logger"
+	"github.com/snapcore/snapd/osutil"
+	"github.com/snapcore/snapd/overlord/devicestate"
+	"github.com/snapcore/snapd/overlord/snapstate"
+	"github.com/snapcore/snapd/overlord/state"
+)
+
+const (
+	optionKernelCmdlineAppend              = "system.kernel.cmdline-append"
+	optionKernelDangerousCmdlineAppend     = "system.kernel.dangerous-cmdline-append"
+	coreOptionKernelCmdlineAppend          = "core.system.kernel.cmdline-append"
```

Comment:
(nitpick³) I guess this could also be:
```go
--- a/overlord/configstate/configcore/kernel.go
+++ b/overlord/configstate/configcore/kernel.go
@@ -38,8 +38,8 @@ import (
 const (
        optionKernelCmdlineAppend              = "system.kernel.cmdline-append"
        optionKernelDangerousCmdlineAppend     = "system.kernel.dangerous-cmdline-append"
-       coreOptionKernelCmdlineAppend          = "core.system.kernel.cmdline-append"
-       coreOptionKernelDangerousCmdlineAppend = "core.system.kernel.dangerous-cmdline-append"
+       coreOptionKernelCmdlineAppend          = "core." + optionKernelCmdlineAppend
+       coreOptionKernelDangerousCmdlineAppend = "core." + optionKernelDangerousCmdlineAppend
 )
 
 func init() {
```

Reply:
Good point, changed now